compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -4551,16 +4551,32 @@ class JSCodeGen()(using genCtx: Context) {
   }
 
   private object WrapArray {
-    lazy val isWrapArray: Set[Symbol] = {
-      val names0 = defn.ScalaValueClasses().map(sym => nme.wrapXArray(sym.name))
-      val names1 = names0 ++ Set(nme.wrapRefArray, nme.genericWrapArray)
-      val symsInPredef = names1.map(defn.ScalaPredefModule.requiredMethod(_))
-      val symsInScalaRunTime = names1.map(defn.ScalaRuntimeModule.requiredMethod(_))
-      (symsInPredef ++ symsInScalaRunTime).toSet
+    lazy val wrapArraySymToToVarArgsName: Map[Symbol, MethodName] = {
+      val SeqClassRef = jstpe.ClassRef(encodeClassName(defn.SeqClass))
+
+      val items: Seq[(Name, String, jstpe.TypeRef)] = Seq(
+        (nme.genericWrapArray, "toGenericVarArgs", jswkn.ObjectRef),
+        (nme.wrapRefArray, "toRefVarArgs", jstpe.ArrayTypeRef(jswkn.ObjectRef, 1)),
+        (tpd.wrapArrayMethodName(defn.UnitType), "toUnitVarArgs", jstpe.ArrayTypeRef(jstpe.ClassRef(jswkn.BoxedUnitClass), 1)),
+        (tpd.wrapArrayMethodName(defn.BooleanType), "toBooleanVarArgs", jstpe.ArrayTypeRef(jstpe.BooleanRef, 1)),
+        (tpd.wrapArrayMethodName(defn.CharType), "toCharVarArgs", jstpe.ArrayTypeRef(jstpe.CharRef, 1)),
+        (tpd.wrapArrayMethodName(defn.ByteType), "toByteVarArgs", jstpe.ArrayTypeRef(jstpe.ByteRef, 1)),
+        (tpd.wrapArrayMethodName(defn.ShortType), "toShortVarArgs", jstpe.ArrayTypeRef(jstpe.ShortRef, 1)),
+        (tpd.wrapArrayMethodName(defn.IntType), "toIntVarArgs", jstpe.ArrayTypeRef(jstpe.IntRef, 1)),
+        (tpd.wrapArrayMethodName(defn.LongType), "toLongVarArgs", jstpe.ArrayTypeRef(jstpe.LongRef, 1)),
+        (tpd.wrapArrayMethodName(defn.FloatType), "toFloatVarArgs", jstpe.ArrayTypeRef(jstpe.FloatRef, 1)),
+        (tpd.wrapArrayMethodName(defn.DoubleType), "toDoubleVarArgs", jstpe.ArrayTypeRef(jstpe.DoubleRef, 1))
+      )
+
+      items.map { case (wrapArrayName, simpleName, argTypeRef) =>
+        val wrapArraySym = defn.getWrapVarargsArrayModule.requiredMethod(wrapArrayName)
+        val toVarArgsName = MethodName(simpleName, argTypeRef :: Nil, SeqClassRef)
+        wrapArraySym -> toVarArgsName
+      }.toMap
     }
 
     def unapply(tree: Apply): Option[Tree] = tree match {
-      case Apply(wrapArray_?, List(wrapped)) if isWrapArray(wrapArray_?.symbol) =>
+      case Apply(wrapArray_?, List(wrapped)) if wrapArraySymToToVarArgsName.contains(wrapArray_?.symbol) =>
```

Comment:
How do the array wrapping methods get introduced? I don't really see them mentioned anywhere else in the diff. Is it already part of the compiler?

P/s: just interested in the rationale behind the `wrapArray_?` name :thinking: